### ISSUE TITLE
OAuth2::AccessToken#get breaks field name on query.

### ISSUE THREAD

**Issue by asm256** [oauth-xx/oauth2#432]
```ruby
cli = OAuth2::Client.new(nil, nil, site: 'https://httpbin.org')
token = OAuth2::AccessToken.new(cli, '0');
params = {"foo[bar][1]" => "val"}

# bad
p token.get("get", params: params).parsed["args"].keys # => ["foo[bar][]"]

# I found a solution
# via: https://github.com/lostisland/faraday/blob/88de44ebe3c40f9050553ce7c0018990b92c2f38/lib/faraday/connection.rb#L135
p token.get("get"){|req| req.params.update(params)}.parsed["args"].keys # => ["foo[bar][1]"]
```

**Comment by pboling** [comment]
Looks like a bug.  If you have bandwidth a failing spec would be great.

**Comment by asm256** [comment]
I am not good at English but I tried it. #433 

ah ... I am still suffering.
Should I also test OAuth2::Client#request ?

**Comment by pboling** [comment]
Thanks @asm256, your fix was released in 1.4.3! ❤️
